src/coreclr/src/jit/lsra.cpp

```diff
@@ -5098,13 +5043,19 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock)
         }
     }
 
-    // Unassign any registers that are no longer live.
+    // Unassign any registers that are no longer live, and set register state, if allocating.
+    if (!allocationPassComplete)
+    {
+        resetRegState();
+        setRegsInUse(liveRegs);
+    }
     for (regNumber reg = REG_FIRST; reg < ACTUAL_REG_COUNT; reg = REG_NEXT(reg))
     {
+        RegRecord* physRegRecord = getRegisterRecord(reg);
```

Comment:
Might want to remove `RegRecord* physRegRecord = getRegisterRecord(reg);` present in `else` part on line 5104 below.